### ISSUE TITLE
Tweak fortios_configuration_fact to adopt gather_subset parity

### ISSUE THREAD

**Issue by ryanmerolle** [fortinet-ansible-dev/ansible-galaxy-fortios-collection#79]
Generally, gather_facts modules in ansible network modules follow a gather_subset mechanism.  It would be good to just follow that same model and gain more core feature parity.  This would allow for native use of gather_facts functionality recently introduced for [network_os devices.](http://blog.networktocode.com/post/ansible-gather_facts-enabled/).  Overall, it feels like thought there are a ton of high quality modules, some minor tweaks to introduce uniformity with the approach of other network modules would be nice.

For example, it would be useful to get the full output from fortios_configuration_fact (i.e. all selectors) in one task in order to generate a full dictionary of a host which could then be run as is against the various modules since the gather_subsets of "all" as well as  Right now you have to run the task multiple times with different selectors to do this.

**Comment by JonTheNiceGuy** [comment]
I ran a quick test against a cloud instance of mine, and I don't think this module does what you're after...

Here's the code I ran: https://gist.github.com/JonTheNiceGuy/6d76802400440646ad3454b157affcc6

Basically what you get back from these is all the config values - it's not "facts" like you'd get from running `get system status` or `get system interface`, it's more akin to running `show system global`.

You're more likely to want the values found with `fortios_facts` - or at least that's what I'd want to see from a `gather_facts` module.

**Comment by ryanmerolle** [comment]
@JonTheNiceGuy this is a good approach that loops a module.  It would be nice to have a function that does not require an ansible loop and frankly follows the approach other network vendor modules take.  Uniformity would allow us to insert fortios into our existing multi vendor playbook without many workarounds like the one you suggested.

**Comment by JieX19** [comment]
Hi @JonTheNiceGuy,

Yes, the fortios_configuration_fact module is basically fetching the config values, it's used for all the configuration APIs not Mornitor APIs. As you mentioned, facts like get_system_status or get_system_interface, they're Mornitor APIs. We are working on it and will introduce this feature in our new release.

**Comment by JieX19** [comment]
Hi @ryanmerolle 

We've released the version 2.0.2 last week. You can get more than one configuration facts in one task now. And check_mode is supported in most of the modules. 

```
  tasks:
  - name: Get multiple selectors info
    fortios_configuration_fact:
      selectors:
        - selector: firewall_address
          params:
            name: "gmail.com"
        - selector: system_interface
        - selector: log_eventfilter
```

Please let us know if you have any questions.

Thanks!
